org.koreader.KOReader.yml

```diff
@@ -0,0 +1,58 @@
+app-id: org.koreader.KOReader
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+base-version: "22.08"
+command: koreader
+separate-locales: false
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --talk-name=org.freedesktop.Notifications
+  - --filesystem=host
+modules:
+  - name: koreader
+    buildsystem: simple
+    sources:
+      - type: file
+        only-arches: [x86_64]
+        url: https://github.com/koreader/koreader/releases/download/v2022.11/koreader-appimage-x86_64-linux-gnu-v2022.11.AppImage
```

Comment:
why no building from source?

Reply:
The project looks very buildable, its documented here: https://github.com/koreader/koreader/blob/master/doc/Building.md

Reply:
In general it works by luck, the CI happens to use an ABI similar to the Flatpak runtime. All it takes is Flatpak or the CI system to have a mismatch and everything breaks, sometimes in subtle ways.

Reply:
The AppImage is purposefully built on Ubuntu 18.04 (perhaps not quite CentOS 7, but it's not, say, Ubuntu 22.04) so presumably much of the luck is manufactured, even if it's still luck. ;-)

Reply:
> While I understand your frustration, it's good enough for me as it's built by CI (not flathub.org, of course). It's actually more maintainable from my perspective, considering the effort of syncing the complicated build script from upstream. Said so, I may revisit it to make it build from source on flathub.org and I always welcome such contribution.

Please give it a try with https://github.com/koreader/koreader-base/pull/1504 and https://github.com/koreader/koreader/pull/9351 merged.

It should be as easy as:

1. clone repo and jump there
2. fetch 3rdparty: `./kodev fetch-thirdparty`
3. build native `./kodev release linux`
4. then you'll end with a tar.gz with everything *except* sdl2
5. use that as a base for flathub recipe

Reply:
@pazos Just to note, you can't "fetch" things during a build, however if they are just submodules they can cloned, or listed in the sources.

Reply:
If you *only* depend on libc ABI, and thats a big if unless you explicitly test this, then you will be OK.

Reply:
>  you can't "fetch" things during a build, however if they are just submodules they can cloned, or listed in the sources.

Well, that's unfortunate.

We rely on CMake scripts to pull different packages from specific git branches/tags, from our own forked git repos or from plain `tar.gz` releases from upstream.

The same set of scripts is in charge of checking, downloading, patching and building each one of the dependencies.

> can you provide a target without git submodule update or other network access? I can patch them out but it's better to include it in the upstream

See 3 from https://github.com/koreader/koreader-base/issues/1486#issuecomment-1160305536

We're not interested in doing more work upstream. That work belongs to package maintainers.

> to my understanding we'll need to git clone each and every packages under koreader-base/thirdparty/ to build the base during build time?

A few more things, including translations, ` lfs` and a few other submodules spreaded across the codebase.

> If that's the case, I don't think it makes sense to replicate that in this PR, unless we can make those packages as submodules under koreader-base.

I'm beginning to agree with you. Is not possible to build from source without *somebody* **not upstream** willing to put the effort to adapt KO build script to the specific requirements of the build environment.

The best thing upstream can do without too much effort is publish a generic linux `tar.gz`  release and flathub can use it as the input for the recipe until somebody steps in and makes possible to build from source on this environment (see https://github.com/NixOS/nixpkgs/pull/178557 for an attempt to build KO from source on a networkless host).